src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,59 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+class FigureSupplier {
+    private static final int FIGURE_COUNT = FigureType.values().length;
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Random random = new Random();
+
+    public Figure getRandomFigure() {
+        FigureType figureType = FigureType.values()[random.nextInt(FIGURE_COUNT)];
+        double[] properties = generateRandomProperties(figureType);
+        String color = colorSupplier.getRandomColor();
+        return createFigure(figureType, properties, color);
+    }
+
+    public Figure getDefaultFigure() {
+        return createFigure(FigureType.CIRCLE, new double[]{10.0}, "white");
+    }
+
+    private double[] generateRandomProperties(FigureType figureType) {
```

Comment:
no reason for this method - delete it.

Better create `private double getRandomLength()`